torch/csrc/jit/runtime/static/passes.cpp

```diff
@@ -812,18 +813,27 @@ void EliminateTrivialEquallySplit(std::shared_ptr<torch::jit::Graph>& graph) {
       continue;
     }
 
-    Node* list_unpack_node = value_out->uses()[0].user;
-    if (list_unpack_node->kind() != prim::ListUnpack) {
+    const Value* value_num_splits = node->inputs()[1];
+
+    const auto num_splits_ivalue = toIValue(value_num_splits);
+
+    // check if equally_split num of splits is 1.
+    if(!num_splits_ivalue.has_value() || !num_splits_ivalue.value().isInt() || num_splits_ivalue.value().toInt() != 1) {
```

Comment:
I think using `constant_as<int>()` would be briefer here (and the other places you use `toIValue`)

above:

`const auto opt_num_splits = constant_as<int>(value_num_splits);`

and here:

`if (opt_num_splits.value_or(0) != 1) {`

Reply:
why is it correct to add this check to the ListUnpack case as well? was it a bug that it wasn't there before? if so, can you split this check out of this PR into a separate PR that comes with a test case demonstrating why it was broken before? If not, aren't we regressing the ListUnpack optimization?